<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>831</width>
    <height>644</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>CoCEL Serial Terminal v1</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>10</y>
      <width>391</width>
      <height>51</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>28</pointsize>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>Serial Terminal</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>80</y>
      <width>41</width>
      <height>21</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>11</pointsize>
      <bold>false</bold>
     </font>
    </property>
    <property name="text">
     <string>PORT</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>110</y>
      <width>71</width>
      <height>21</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>11</pointsize>
      <bold>false</bold>
     </font>
    </property>
    <property name="text">
     <string>Baud Rate</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
    </property>
   </widget>
   <widget class="QComboBox" name="comboBox_port">
    <property name="geometry">
     <rect>
      <x>90</x>
      <y>80</y>
      <width>121</width>
      <height>22</height>
     </rect>
    </property>
   </widget>
   <widget class="QComboBox" name="comboBox_baudrate">
    <property name="geometry">
     <rect>
      <x>90</x>
      <y>110</y>
      <width>121</width>
      <height>22</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>9600</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>115200</string>
     </property>
    </item>
   </widget>
   <widget class="QPushButton" name="pushButton_connect">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>220</x>
      <y>80</y>
      <width>71</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string>Connect</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_disconnect">
    <property name="geometry">
     <rect>
      <x>300</x>
      <y>80</y>
      <width>71</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string>Disconnect</string>
    </property>
   </widget>
   <widget class="QGroupBox" name="groupBox_transmit">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>140</y>
      <width>811</width>
      <height>161</height>
     </rect>
    </property>
    <property name="title">
     <string>Transmit</string>
    </property>
    <widget class="QTextBrowser" name="textBrowser_trans">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>50</y>
       <width>711</width>
       <height>101</height>
      </rect>
     </property>
     <property name="verticalScrollBarPolicy">
      <enum>Qt::ScrollBarAsNeeded</enum>
     </property>
     <property name="horizontalScrollBarPolicy">
      <enum>Qt::ScrollBarAsNeeded</enum>
     </property>
    </widget>
    <widget class="QLineEdit" name="lineEdit_data_transmit">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>711</width>
       <height>22</height>
      </rect>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_send">
     <property name="geometry">
      <rect>
       <x>730</x>
       <y>20</y>
       <width>71</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Send</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_stop_send">
     <property name="geometry">
      <rect>
       <x>730</x>
       <y>60</y>
       <width>71</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Stop</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_clear_trans">
     <property name="geometry">
      <rect>
       <x>730</x>
       <y>100</y>
       <width>71</width>
       <height>51</height>
      </rect>
     </property>
     <property name="text">
      <string>Clear</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_2">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>300</y>
      <width>811</width>
      <height>161</height>
     </rect>
    </property>
    <property name="title">
     <string>Receive</string>
    </property>
    <widget class="QTextBrowser" name="textBrowser_receive">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>711</width>
       <height>131</height>
      </rect>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_clear_receive">
     <property name="geometry">
      <rect>
       <x>730</x>
       <y>20</y>
       <width>71</width>
       <height>51</height>
      </rect>
     </property>
     <property name="text">
      <string>Clear</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_3">
    <property name="geometry">
     <rect>
      <x>620</x>
      <y>70</y>
      <width>201</width>
      <height>71</height>
     </rect>
    </property>
    <property name="title">
     <string>Mode</string>
    </property>
    <widget class="QRadioButton" name="radioButton_mode_normal">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>61</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>Normal</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
     <property name="autoRepeat">
      <bool>false</bool>
     </property>
    </widget>
    <widget class="QRadioButton" name="radioButton_mode_repeat">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>40</y>
       <width>91</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Auto Repeat</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="lineEdit_hz">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>40</y>
       <width>71</width>
       <height>22</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="label_4">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>40</y>
       <width>21</width>
       <height>21</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>9</pointsize>
       <bold>false</bold>
      </font>
     </property>
     <property name="text">
      <string>Hz</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
   </widget>
   <widget class="Line" name="line">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>460</y>
      <width>831</width>
      <height>21</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QGroupBox" name="groupBox_4">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>480</y>
      <width>381</width>
      <height>141</height>
     </rect>
    </property>
    <property name="title">
     <string>Log</string>
    </property>
    <widget class="QPushButton" name="pushButton_clear_log">
     <property name="geometry">
      <rect>
       <x>300</x>
       <y>20</y>
       <width>71</width>
       <height>51</height>
      </rect>
     </property>
     <property name="text">
      <string>Clear</string>
     </property>
    </widget>
    <widget class="QTextBrowser" name="textBrowser_log">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>281</width>
       <height>111</height>
      </rect>
     </property>
    </widget>
   </widget>
   <widget class="QPushButton" name="pushButton_close">
    <property name="geometry">
     <rect>
      <x>760</x>
      <y>10</y>
      <width>61</width>
      <height>51</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: rgb(255, 0, 0);
color: rgb(255, 255, 255);</string>
    </property>
    <property name="text">
     <string>Close</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>420</x>
      <y>10</y>
      <width>331</width>
      <height>51</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>9</pointsize>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>Copyright (C) 2024 Sanghyun Park. All right reserved.
CoCEL @POSTECH
Contact: pash0302@gmail.com</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
    </property>
   </widget>
   <widget class="QComboBox" name="comboBox_dataBits">
    <property name="geometry">
     <rect>
      <x>520</x>
      <y>81</y>
      <width>81</width>
      <height>21</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>8</string>
     </property>
    </item>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>450</x>
      <y>80</y>
      <width>71</width>
      <height>21</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>11</pointsize>
      <bold>false</bold>
     </font>
    </property>
    <property name="text">
     <string>Data bits</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_7">
    <property name="geometry">
     <rect>
      <x>470</x>
      <y>110</y>
      <width>51</width>
      <height>21</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>11</pointsize>
      <bold>false</bold>
     </font>
    </property>
    <property name="text">
     <string>Parity</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QComboBox" name="comboBox_parity">
    <property name="geometry">
     <rect>
      <x>520</x>
      <y>110</y>
      <width>81</width>
      <height>22</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>no parity</string>
     </property>
    </item>
   </widget>
   <widget class="QPushButton" name="pushButton_refreshPort">
    <property name="geometry">
     <rect>
      <x>380</x>
      <y>80</y>
      <width>71</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string>Refresh
PORT</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>831</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
